### PR TITLE
CAPZ: use main branch for dual stack jobs

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
This PR temporarily sets the CAPZ branch to main for ipv6 and dual stack jobs that were recently added since https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4086 is not yet part of a CAPZ release branch.

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh [hunk 1]
```diff
@@ -375,7 +375,7 @@ periodics:
   extra_refs:
   - org: kubernetes-sigs
     repo: cluster-api-provider-azure
-    base_ref: ${capz_periodic_branch_name}
+    base_ref: main # TODO: switch back to capz_periodic_branch_name once capz release-1.12 exists
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/generate.sh:378]: alternative would be to backport the PR but since it's quite large and touches template this is the quickest way to get signal. release-1.12 should be available mid November, at which point we can revert this change.

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 2]
```diff
@@ -336,7 +336,7 @@ periodics:
   extra_refs:
   - org: kubernetes-sigs
     repo: cluster-api-provider-azure
-    base_ref: main
+    base_ref: main # TODO: switch back to capz_periodic_branch_name once capz release-1.12 exists
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:339]: this was already testing at main but it's auto-generated by the generate script

**Review by jackfrancis — APPROVED** [review]
/lgtm

I think this is the most practical approach to unblock the most number of projects
